packages/react-native-reanimated/src/layoutReanimation/web/transition/EntryExit.web.ts

```diff
@@ -0,0 +1,52 @@
+'use strict';
+import type { AnimationData, TransitionData } from '../animationParser';
+import { AnimationsData } from '../config';
+
+export function EntryExitTransition(
+  name: string,
+  transitionData: TransitionData
+) {
+  const exitingAnimationData = structuredClone(
+    AnimationsData[transitionData.exiting]
+  );
+  const enteringAnimationData = structuredClone(
+    AnimationsData[transitionData.entering]
+  );
+
+  const additionalData = {
+    translateX: `${transitionData.translateX}px`,
+    translateY: `${transitionData.translateY}px`,
+    scale: `${transitionData.scaleX},${transitionData.scaleY}`,
+  };
+
+  const keyframeData: AnimationData = {
+    name,
+    style:
+      '0' in exitingAnimationData.style
+        ? {}
+        : { 0: { transform: [additionalData] } },
+    duration: 300,
+  };
+
+  for (const [timestamp, styles] of Object.entries(
+    exitingAnimationData.style
+  )) {
+    if (styles.transform !== undefined) {
+      styles.transform.unshift(additionalData);
+    } else {
+      styles.transform = [additionalData];
+    }
```

Comment:
Let's add comment or move it to another function

Reply:
Done in [026c08f](https://github.com/software-mansion/react-native-reanimated/pull/6232/commits/026c08f5e422fd8a8443d20bed2d4bb8f3b2391c)